src/rpc/response.zig

```diff
@@ -49,13 +48,13 @@ pub fn Response(comptime Method: type) type {
             };
         }
 
-        pub fn deinit(self: Response(Method)) void {
+        pub fn deinit(self: Self) void {
```

Comment:
Did you switch to `Self` because it's a generic type? I don't know much about the internals of the compiler, but it's proven to be smart about avoiding infinite recursion here. So I'm not clear on the purpose of `Self` in this case.

Reply:
I switched it to `Self` since that's consistent with the rest of the code base, and is what we have codified in the style guide: https://github.com/Syndica/sig/blob/main/docs/docusaurus/docs/contributing/style-guide.mdx#method-parameters

Reply:
Actually the style guide says the opposite. Take a closer look, it says to use the struct name. Not sure if you remember the discussion we had about this a few months ago when reviewing some of David's code in the VM,  and we decided to start using this pattern. It's not a big deal, and we don't need to change the entire codebase right now, but I was a little confused by your change because it goes in the opposite direction. I thought you were making an exception because it was a generic type, and wanted to understand that a little better.

Reply:
I'm referring to:
> If the type name is not available (for example in anonymous structs), define const Self = @This() and use that as the type.

I interpret "anonymous struct" refers to structs that aren't directly assigned a name in global scope, ie anonymous structs. If that isn't the intention, we should clarify that.

Personally, generic types are the exact one place where I think `const Self = @This()`, `const Name = @This()`, or even `@This()` is preferable, given that otherwise changing the parameters in any way leads to more changes than would otherwise be needed during refactoring (certainly not as simple as renaming the struct).

Do you want to keep it as is, or go forward with the change?

Reply:
We can leave this as is since it's undecided right now and not important enough to block the merge.

My intended meaning of "name is not available" was that there is simply no name that can be given to the type. `Response(Method)` is a name that can be given to the type, so it does not qualify, according to my intention. We can clarify this language once we settle on a standard.